autoload/arduino.vim

```diff
@@ -648,6 +653,12 @@ function! arduino#GetPorts() abort
       call add(ports, port)
     endfor
   endfor
+  if s:OS ==? 'Windows'
+    let found = split(system('pwsh -nop -c "arduino-cli board list | Select-String -Pattern \"^(COM\d) \" | ForEach-Object { $_.Matches.Value }"'), '\n')
+    for port in found
+      call add(ports, port)
+    endfor
+  endif 
```

Comment:
Would prefer if you could call `arduino-cli` directly so it doesn't rely on powershell. Do the ports appear in the json data if you run it like we do here?
https://github.com/stevearc/vim-arduino/blob/111db616db21d4f925691f1517792953f7671647/autoload/arduino.vim#L251

If not, you could still just call `arduino-cli board list` and the extract the information you want using `matchlist()`.

Reply:
This makes sense. I'm gonna change it.